plrust/src/allow_list.rs

```diff
@@ -212,6 +212,53 @@ pub enum Error {
     UnsupportedVersionReq(String),
 }
 
+// Tuple describes name of the dependency, version of the dependency, features of the dependecy that are enabled, whether default features enabled
+pub type AllowedDependencyTuple = (String, String, Vec<String>, bool);
+
+// Convert all the allowed dependencies into a vector of tuples
+pub fn get_allowed_dependencies_as_tuple_vector() -> Vec<AllowedDependencyTuple> {
+    let allowlist = load_allowlist().expect("Error loading dependency allow-list");
+    let mut tuples: Vec<AllowedDependencyTuple> = vec![];
+
+    for dependency in allowlist.values() {
+        let mut tuple: Vec<AllowedDependencyTuple> =
+            get_single_dependency_as_tuple_vector(dependency);
+        tuples.append(&mut tuple);
+    }
+    tuples
+}
+
+// Multiple version of a dependencies can be allowed. Create a tuple for each version of the dependency
+fn get_single_dependency_as_tuple_vector(dep: &Dependency) -> Vec<AllowedDependencyTuple> {
+    let mut tuples: Vec<AllowedDependencyTuple> = vec![];
+    for version in dep.versions.values() {
+        let name = dep.name.clone();
+        let ver = version
+            .get("version")
+            .unwrap()
+            .to_string()
+            .replace("\"", "");
+        let features: Vec<String> = match version.get("features") {
+            Some(features) => match features.is_array() {
+                true => features
+                    .as_array()
+                    .unwrap()
```

Comment:
Same here, if it's possible that "features" won't be an array, then we should for this situation in the data structure instead of panicking.

Reply:
If i understand correctly it should not panic here, because before calling `unwrap`,  i have already first checked if the field `features` exists, and if so whether the value for the field `feature` is an array, only then i would unwrap the value.

Reply:
Actually, the code should simply use `as_array().unwrap()` instead of first checking `is_array()`, because if there is an entry, it is only valid for it to be an array.

Reply:
yea ideally it should be an array, but apparently i think the code that parses the toml file does not check if the value is an array of not

https://github.com/tcdi/plrust/blob/main/plrust/src/allow_list.rs#L218

Reply:
Actually on second thoughts, i have added some code to validate that features is any array of string

Reply:
Yes, I am fine with addressing the larger concern re: that file in another PR. I just think we should be locking that down. The entries should more-or-less match what a Cargo.toml accepts, to minimize the principle of surprise.

Reply:
Ah, I see you did some of the changes already. Thank you!